sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -207,31 +207,32 @@ private[sbtplugin] object ScalaJSPluginInternal {
               case Stage.FullOpt => "Full"
             }
 
-            log.info(s"$stageName optimizing $output")
+            log.info(s"$stageName optimizing $outputDir")
 
-            IO.createDirectory(output.getParentFile)
+            IO.createDirectory(outputDir)
 
-            def relURI(path: String) = new URI(null, null, path, null)
-
-            val out = LinkerOutput(linkerImpl.outputFile(output.toPath))
-              .withSourceMap(linkerImpl.outputFile(sourceMapFile.toPath))
-              .withSourceMapURI(relURI(sourceMapFile.getName))
-              .withJSFileURI(relURI(output.getName))
+            val out = OutputSpec(linkerImpl.outputDirectory(outputDir.toPath))
+            val moduleSpec = List(
+              ModuleSpec(modName)
+                .withInitializers(moduleInitializers)
+                .withSelector(ModuleSpec.Selector.default)
+            )
```

Comment:
sbt plugin needs a lot more work. It's not 100% clear to me ATM how we can reasonably support existing builds that assume single-file outputs without introducing a lot of double logic.